core/src/data/value/mod.rs

```diff
@@ -555,6 +555,45 @@ impl Value {
     pub fn to_cmp_be_bytes(&self) -> Result<Vec<u8>> {
         self.try_into().map(|key: Key| key.to_cmp_be_bytes())
     }
+
+    /// only Value::Str type allowed
+    /// returns positon of substring from string
+    /// if there are substring in string, returns the position where the first letter of the substring begins. The minimum value is 1
+    /// else returns 0 because there is no substring to find within the string
+    /// # Examples
+    /// ```
+    /// use gluesql_core::data::{ValueError, value::Value::*};
+    ///
+    /// let str1 = Str("ramen".to_string());
+    /// let str2 = Str("men".to_string());
+    /// let empty_str = Str("".to_string());
+    /// assert_eq!(str1.str_position(&str2), Ok(3));
+    /// assert_eq!(str2.str_position(&str1), Ok(0));
+    /// assert_eq!(Null.str_position(&str2), Ok(0));
+    /// assert_eq!(str1.str_position(&Null), Ok(0));
+    /// assert_eq!(empty_str.str_position(&str2), Ok(0));
+    /// assert_eq!(str1.str_position(&empty_str), Ok(0));
+    /// assert_eq!(
+    ///     str1.str_position(&I8(1)),
+    ///     Err(ValueError::StrPositionOnNonString(str1, I8(1)).into())
+    /// );
+    /// ```
```

Comment:
<https://doc.rust-lang.org/stable/std/primitive.str.html#method.find>
I rewrote it by referring to the reference. Apply markdown lint rules.

```suggestion
    /// Support only [`Value::Str`] variant
    /// 
    /// Returns the position where the first letter of the substring starts if the string contains a substring.
    /// 
    /// Returns `0` if the string to be found is not found.
    /// Returns minimum value `1` when the string is found.
    /// 
    /// # Examples
    /// 
    /// ```
    /// use gluesql::core::data::{ValueError, value::Value::*};
    ///
    /// let str1 = Str("ramen".to_string());
    /// let str2 = Str("men".to_string());
    /// let empty_str = Str("".to_string());
    /// assert_eq!(str1.str_position(&str2), Ok(3));
    /// assert_eq!(str2.str_position(&str1), Ok(0));
    /// assert_eq!(Null.str_position(&str2), Ok(0));
    /// assert_eq!(str1.str_position(&Null), Ok(0));
    /// assert_eq!(empty_str.str_position(&str2), Ok(0));
    /// assert_eq!(str1.str_position(&empty_str), Ok(0));
    /// assert_eq!(
    ///     str1.str_position(&I8(1)),
    ///     Err(ValueError::StrPositionOnNonString(str1, I8(1)).into())
    /// );
    /// ```
```
<img width="1031" alt="image" src="https://user-images.githubusercontent.com/27040963/192131462-1af5579c-f35c-4959-b3d2-8460b6bad770.png">